openai/openai_test.go

```diff
@@ -0,0 +1,114 @@
+package openai
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/gin-gonic/gin"
+	"github.com/ollama/ollama/api"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMiddleware(t *testing.T) {
+	type testCase struct {
+		Name       string
+		Method     string
+		Path       string
+		TestPath   string
+		Middleware func() gin.HandlerFunc
+		Endpoint   func(c *gin.Context)
+		Setup      func(t *testing.T, req *http.Request)
+		Expected   func(t *testing.T, resp *httptest.ResponseRecorder)
+	}
+
+	testCases := []testCase{
+		{
+			Name:       "OpenAI Chat Handler",
+			Method:     http.MethodPost,
+			Path:       "/api/chat",
+			TestPath:   "/api/chat",
+			Middleware: Middleware,
+			Endpoint: func(c *gin.Context) {
+				c.JSON(http.StatusOK, api.ChatResponse{
+					Message: api.Message{
+						Role:    "assistant",
+						Content: "Hello!",
+					},
+				})
+			},
+			Setup: func(t *testing.T, req *http.Request) {
+				body := ChatCompletionRequest{
+					Model:    "test-model",
+					Messages: []Message{{Role: "user", Content: "Hello"}},
+				}
+
+				bodyBytes, _ := json.Marshal(body)
+
+				req.Body = io.NopCloser(bytes.NewReader(bodyBytes))
+				req.Header.Set("Content-Type", "application/json")
+			},
+			Expected: func(t *testing.T, resp *httptest.ResponseRecorder) {
+				var chatResp ChatCompletion
+				if err := json.NewDecoder(resp.Body).Decode(&chatResp); err != nil {
+					t.Fatal(err)
+				}
+
+				assert.Equal(t, "chat.completion", chatResp.Object)
+				assert.Equal(t, "Hello!", chatResp.Choices[0].Message.Content)
+			},
+		},
+		{
+			Name:       "OpenAI List Handler",
```

Comment:
we don't need `OpenAI ` in here since it's already in `openai_test.go`, just `list handler` is great